src/SQLStore/PropertyTableInfoFetcher.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+namespace SMW\SQLStore;
+
+use SMW\DataTypeRegistry;
+use SMW\DIProperty;
+use SMWDataItem as DataItem;
+
+/**
+ * @license GNU GPL v2+
+ * @since 2.2
+ *
+ * @author mwjames
+ */
+class PropertyTableInfoFetcher {
+
+	/**
+	 * Array for keeping property table table data, indexed by table id.
+	 * Access this only by calling getPropertyTables().
+	 *
+	 * @var TableDefinition[]|null
+	 */
+	private $propertyTableDefinitions = null;
+
+	/**
+	 * Array to cache "propkey => table id" associations for fixed property
+	 * tables. Initialized by getPropertyTables(), which must be called
+	 * before accessing this.
+	 *
+	 * @var array|null
+	 */
+	private $fixedPropertyTableIds = null;
+
+	/**
+	 * Keys of special properties that should have their own
+	 * fixed property table.
+	 *
+	 * @var array
+	 */
+	private $customizableSpecialProperties = array(
+		'_MDAT', '_CDAT', '_NEWP', '_LEDT', '_MIME', '_MEDIA',
+	);
+
+	/**
+	 * @var array
+	 */
+	private $customSpecialPropertyList = array();
+
+	/**
+	 * @var array
+	 */
+	private $fixedSpecialProperties = array(
+		// property declarations
+		'_TYPE', '_UNIT', '_CONV', '_PVAL', '_LIST', '_SERV',
+		// query statistics (very frequently used)
+		'_ASK', '_ASKDE', '_ASKSI', '_ASKFO', '_ASKST', '_ASKDU',
+		// subproperties, classes, and instances
+		'_SUBP', '_SUBC', '_INST',
+		// redirects
+		'_REDI',
+		// has sub object
+		'_SOBJ',
+		// vocabulary import and URI assignments
+		'_IMPO', '_URI',
+		// Concepts
+		'_CONC'
+	);
+
+	/**
+	 * @var array
+	 */
+	private $customFixedPropertyList = array();
+
+	/**
+	 * Default tables to use for storing data of certain types.
+	 *
+	 * @var array
+	 */
+	private $defaultDiTypeTableIdMap = array(
+		DataItem::TYPE_NUMBER     => 'smw_di_number',
+		DataItem::TYPE_BLOB       => 'smw_di_blob',
+		DataItem::TYPE_BOOLEAN    => 'smw_di_bool',
+		DataItem::TYPE_URI        => 'smw_di_uri',
+		DataItem::TYPE_TIME       => 'smw_di_time',
+		DataItem::TYPE_GEO        => 'smw_di_coords', // currently created only if Semantic Maps are installed
+		DataItem::TYPE_WIKIPAGE   => 'smw_di_wikipage',
+		//DataItem::TYPE_CONCEPT    => '', // _CONC is the only property of this type
+	);
+
+	/**
+	 * @since 2.2
+	 *
+	 * @param array $customFixedProperties
+	 */
+	public function setCustomFixedPropertyList( array $customFixedProperties ) {
```

Comment:
In this case I renamed them to something like `setEnabledCustomSpecialPropertyList`